src/platform/OpenThread/GenericThreadStackManagerImpl_OpenThread.hpp

```diff
@@ -1255,32 +1267,42 @@ CHIP_ERROR GenericThreadStackManagerImpl_OpenThread<ImplClass>::_WriteThreadNetw
     break;
 
     case ThreadNetworkDiagnostics::Attributes::PartitionId::Id: {
-        uint32_t partitionId = otThreadGetPartitionId(mOTInst);
-        err                  = encoder.Encode(partitionId);
+        err = encoder.EncodeNullableOnCondition(mIsAttached, [this](const auto pAttrEncoder) -> CHIP_ERROR {
+            uint32_t partitionId = otThreadGetPartitionId(mOTInst);
+            return pAttrEncoder->Encode(partitionId);
+        });
     }
     break;
 
     case ThreadNetworkDiagnostics::Attributes::Weighting::Id: {
-        uint8_t weight = otThreadGetLeaderWeight(mOTInst);
-        err            = encoder.Encode(weight);
+        err = encoder.EncodeNullableOnCondition(mIsAttached, [this](const auto pAttrEncoder) -> CHIP_ERROR {
```

Comment:
Here's a question.  Are there _any_ things in this switch that don't depend on mIsAttached?  And if not, could we just check for it up front?

Reply:
A bunch of attributes are not nullable and with return counter values of the thread instance. So I would say no it it's not at large but still I'll do it differently with a starting. I'll do that tomorrow morning

Reply:
Let's just make sure we don't null some attributes that don't need nulling. Some of the offending attributes in the original issue (e.g. ActiveTimestamp) require an active dataset.

Some others (e.g. channel) rely on being configured (active dataset) or "operational".

Reply:
So except for the RoutingRole that can always fetch a relevant state from the thread instance, all other nullable attributes depend on an active dataset provided to the device. If that dataset is commissioned, even if the thread radio is deactivated or not yet attached, relevant data can be fetched.

If the thread instance doesn't have a valid dataset, then all those attributes will read as null.


Bottom line, whatever the attachment state of the device (operational or not), if the relevant data is fetchable (and valid) return that, else return null.